todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModel.java

```diff
@@ -169,15 +156,15 @@ public void completeTask(Task task, boolean completed) {
         return mDataLoading;
     }
 
-    public LiveData<String> getCurrentFilteringLabel() {
+    public MutableLiveData<Integer> getCurrentFilteringLabel() {
         return mCurrentFilteringLabel;
     }
 
-    public LiveData<String> getNoTasksLabel() {
+    public MutableLiveData<Integer> getNoTasksLabel() {
         return mNoTasksLabel;
     }
 
-    public LiveData<Drawable> getNoTaskIconRes() {
+    public MutableLiveData<Integer> getNoTaskIconRes() {
         return mNoTaskIconRes;
     }
```

Comment:
@JoseAlcerreca Is it really needed to expose these objects as _Mutable_ LiveData?

Reply:
No, they should be MutableLiveDatas only in the ViewModel, exposed as LiveDatas. That's a bug.